HEC.FDA.Model/compute/ImpactAreaScenarioSimulation.cs

```diff
@@ -124,18 +127,62 @@ public ImpactAreaScenarioResults Compute(IProvideRandomNumbers randomProvider, C
             CreateHistogramsForAssuranceOfThresholds();
             MessageEventArgs beginComputeMessageArgs = new(new Message($"EAD and performance compute for the impact area with ID {_ImpactAreaID} has been initiated" + Environment.NewLine));
             ReportMessage(this, beginComputeMessageArgs);
-            ComputeIterations(convergenceCriteria, randomProvider, masterseed, computeWithDamage, computeIsDeterministic, cancellationToken);
+            ComputeIterations(convergenceCriteria, computeWithDamage, computeIsDeterministic, cancellationToken);
             _ImpactAreaScenarioResults.ParallelResultsAreConverged(.95, .05);
             MessageEventArgs endComputeMessageArgs = new(new Message($"EAD and performance compute for the impact area with ID {_ImpactAreaID} has completed successfully" + Environment.NewLine));
             ReportMessage(this, endComputeMessageArgs);
             return _ImpactAreaScenarioResults;
         }
 
+        private void PopulateRandomNumbers(ConvergenceCriteria convergenceCriteria)
```

Comment:
I'm interested in whether it makes sense to have each individual object track their seeds, or for ImpactAreaScenarioSimulation to track them for them and inject the right one.  What were your thoughts on that?

Reply:
resolved